src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -371,16 +371,45 @@ protected string applyGoTemplateText(string template, Dictionary<string, object>
                 if (condition.StartsWith("."))
                 {
                     var conditionResultState = false;
-                    var value = variables[condition];
 
-                    if (value == null)
-                        conditionResultState = false;
-                    else if (value is string)
-                        conditionResultState = !string.IsNullOrWhiteSpace((string)value);
-                    else if (value is ICollection)
-                        conditionResultState = ((ICollection)value).Count > 0;
+                    var equalityRegex = new Regex(@"(.+)+\s+(eq|ne)\s+(.+)");
+                    var equalityMatches = equalityRegex.Match(condition);
+                    if (equalityMatches.Success)
+                    {
+                        var leftOpVal = equalityMatches.Groups[1].Value;
+                        var eqOpVal = equalityMatches.Groups[2].Value;
+                        var rightOpVal = equalityMatches.Groups[3].Value;
+                        var value = variables[leftOpVal];
+                        switch (eqOpVal)
```

Comment:
Your switch does not match `if (value == null)` and probably should check for that explicitly instead of throwing an exception.